test/units/plugins/connection/test_connection.py

```diff
@@ -135,7 +137,16 @@ def test_lxc_connection_module(self):
         self.assertIsInstance(LxcConnection(self.play_context, self.in_stream), LxcConnection)
 
     def test_local_connection_module(self):
-        self.assertIsInstance(LocalConnection(self.play_context, self.in_stream), LocalConnection)
+        local = LocalConnection(self.play_context, self.in_stream)
+        self.assertIsInstance(local, LocalConnection)
+
+        # Ensure that file paths are encoded properly
+        tempdir = tempfile.mkdtemp()
+        local.cwd = tempdir.encode('utf-8')  # action plugin sets the cwd as 'bytes'
```

Comment:
Have you bisected the change?

Reply:
I'm afraid I don't understand what you mean. This code is an addition to an existing unit test to verify that LocalConnection.put_file now works.

If you were asking if I have bisected from which exact commit the bug originates, then no. I did not check the root cause as in my opinion the fix should anyway be done in _ensure_abs.